src/main/java/br/com/academiadev/thunderpets/controller/PetController.java

```diff
@@ -32,8 +43,24 @@
     }
 
     @PostMapping("/")
-    public Pet salvar(@RequestBody Pet pet) {
-        return petRepository.saveAndFlush(pet);
+    public Pet salvar(@RequestBody PetDTO petDTO) {
+        Localizacao localizacao = localizacaoRepository.saveAndFlush(petDTO.getLocalizacao());
+
+        Pet pet = new Pet(petDTO.getId(), petDTO.getNome(),
+                petDTO.getDescricao(), petDTO.getDataAchado(),
+                petDTO.getDataRegistro(), petDTO.getEspecie(),
+                petDTO.getPorte(), petDTO.getSexo(), petDTO.getStatus(),
+                petDTO.getIdade(), petDTO.getUsuario(),
+                localizacao, petDTO.isAtivo());
+
+        petRepository.saveAndFlush(pet);
```

Comment:
Você tá dando `saveAndFlush` mas não tá pegando o retorno. Assim, quanto você faz o `foto.setPet(pet)` ali embaixo você tá colocando na foto um pet que ainda não tem ID. Isso vai dar ruim na hora que ele for salvar, porque ele precisa do ID do pet pra cadastrar a foto.

Reply:
O pet no qual eu fiz a construção já não possui o ID que eu quero pra salvar as fotos? A questão de pegar o retorno do saveAndFlush é pra eu ter a garantia do ID ou algo do tipo? Ou porque o vai ser gerado um ID novo?